### PR TITLE
Core 2.0 dependencies upgrade: Gradle, AGP, build files

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR upgrades various dependencies for the Android Core 2.0 upgrade:
* AGP: 4.0.1 -> 7.3.1
* Java `VERSION_1_7` -> `VERSION_1_8`
* Gradle 6.1.1 -> 7.5

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/app/build.gradle [hunk 1]
```diff
@@ -38,6 +38,7 @@ android {
     kotlinOptions {
         jvmTarget = '1.8'
     }
+    namespace 'com.adobe.marketing.mobile.edge.bridge.testapp'
```
Comment (kevinlind) [inline code/app/build.gradle:41]: nit: move to the top of the `android` block, just for visibility sake.

File: code/edgebridge/build.gradle [hunk 2]
```diff
@@ -63,6 +61,7 @@ android {
         sourceCompatibility rootProject.ext.sourceCompatibility
         targetCompatibility rootProject.ext.targetCompatibility
     }
+    namespace 'com.adobe.marketing.mobile.edge.bridge'
```
Comment (kevinlind) [inline code/edgebridge/build.gradle:64]: nit: move to the top of the `android` block for visibility.

File: code/edgebridge/build.gradle [hunk 3]
```diff
@@ -46,7 +45,6 @@ android {
     buildTypes {
         debug {
             testCoverageEnabled true
-            debuggable true
```
Comment (kevinlind) [inline code/edgebridge/build.gradle:49]: Can you keep this parameter?  I realize the AGP Migration Assistant removes it stating it is ignored, but I've also read documentation that it is needed to retain the debugging symbols of the library. To be safe, let's keep it in the build configuration.

**Comment by kevinlind** [comment]
In the `build.gradle` `android` block, can you rename the version keys to the following: `compileSDK`, `targetSDK`, `minSDK`?

**Comment by kevinlind** [comment]
Please move the `buildScripts` and `allProjects` repository declaration blocks from `code/build.gradle` to `settings.gradle`.
Here's an example: https://git.corp.adobe.com/dms-mobile/platform-extension/blob/dev-2.0.0/code/settings.gradle

**Comment by kevinlind** [comment]
In `code/edgebridge/build.gradle`, please remove the `publishing.publications.aar` block, leaving only the `publishing.publications.release` block. The `aar` publication is a duplicate of `release` and is not needed.

File: .gitignore [hunk 4]
```diff
@@ -5,6 +5,7 @@
 bin/
 build/
 ci/
+jacoco.exec
 
 # IntelliJ
```
Comment (cacheung) [inline .gitignore:10]: Do you need to add more to the gitignore?
At least 
```
# Local configuration file (sdk path, etc)
local.properties
```

I was looking at Edge and few more can be added.

File: .gitignore [hunk 5]
```diff
@@ -5,6 +5,7 @@
 bin/
 build/
 ci/
+jacoco.exec
```
Comment (kevinlind) [inline .gitignore:8]: Can you also remove the `code/edgebridge/jacoco.exec` file from source control as well?

File: Makefile [hunk 6]
```diff
@@ -43,7 +43,7 @@ ci-build-app:
 
 ci-unit-test: create-ci
 	(mkdir -p ci/unit-test)
-	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) platformUnitTestJacocoReport)
+	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) platformUnitTestJacocoReport --debug)
```
Comment (kevinlind) [inline Makefile:46]: Remove the `--debug` flag.

File: Makefile [hunk 7]
```diff
@@ -55,6 +55,7 @@ ci-functional-test: create-ci
 ci-javadoc: create-ci
 	(mkdir -p ci/javadoc)
 	(./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) javadocPublic > ci/javadocPublic.log 2>&1)
+	# (./code/gradlew -p code/$(EXTENSION-LIBRARY-FOLDER-NAME) javadocPublic --debug 2>&1 | tee -a ci/javadocPublic.log)
```
Comment (kevinlind) [inline Makefile:58]: Remove this commented out code.

File: code/edgebridge/build.gradle [hunk 8]
```diff
@@ -68,14 +69,15 @@ android {
...
-        ext.androidJar = "${android.sdkDirectory}/platforms/${android.compileSdkVersion}/android.jar"
+        ext.androidJar = "${android.sdkDirectory}/platforms/${android.compileSdk}/android.jar"
 
         doFirst{classpath = files(variant.javaCompile.classpath.files) + files(ext.androidJar)}
 
         exclude "**/BuildConfig.java"
         exclude "**/R.java"
         options {
             links "https://developer.android.com/reference"
+            linksOffline "https://d.android.com/reference/", "https://d.android.com/reference/androidx/"
```
Comment (kevinlind) [inline code/edgebridge/build.gradle:80]: Is this line just replacing `https://d.android.com/reference/` with `https://d.android.com/reference/androidx/` when linking offline?  I think this won't work if offline as the URI you are using needs network access.

File: code/edgebridge/build.gradle [hunk 9]
```diff
@@ -208,8 +179,9 @@ task platformUnitTestJacocoReport(type: JacocoReport, dependsOn: "testPhoneDebug
     additionalClassDirs.setFrom files([debugTree])
     additionalSourceDirs.setFrom files(android.sourceSets.main.java.sourceFiles)
     sourceDirectories.setFrom files(android.sourceSets.phone.java.sourceFiles)
-    executionData "$buildDir/jacoco/testPhoneDebugUnitTest.exec"
-
+//    executionData "$buildDir/jacoco/testPhoneDebugUnitTest.exec"
+    executionData "$buildDir/outputs/unit_test_code_coverage/phoneDebugUnitTest/testPhoneDebugUnitTest.exec"
+//    /Users/timothyk/git/timkimadobe/aepsdk-edgebridge-android/code/edgebridge/build/outputs/unit_test_code_coverage/phoneDebugUnitTest
```
Comment (kevinlind) [inline code/edgebridge/build.gradle:184]: Is this creating the coverage reports for you?  I noticed the Core project is using this path as well.
Please remove the commented out lines of code.

File: code/settings.gradle [hunk 10]
```diff
@@ -1,3 +1,43 @@
...
+        // in the individual module build.gradle files
+    }
+}
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        mavenCentral()
+//        maven { url "https://plugins.gradle.org/m2/" }
```
Comment (kevinlind) [inline code/settings.gradle:34]: Remove the comment.

File: code/settings.gradle [hunk 11]
```diff
@@ -1,3 +1,43 @@
...
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+    }
+    dependencies {
+        classpath "com.android.tools.build:gradle:7.3.1"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath "com.diffplug.spotless:spotless-plugin-gradle:$spotless_version"
+        // NOTE: Do not place your application dependencies here; they belong
+        // in the individual module build.gradle files
+    }
+}
```
Comment (kevinlind) [inline code/settings.gradle:27]: Remove this block from `settings.gradle`. Instead, define the dependencies in the `plugins` block in `code/build.gradle`
```
plugins {
    // Android Gradle Plugin version
    id 'com.android.application' version '7.3.1' apply false
    id 'com.android.library' version '7.3.1' apply false
    id 'com.diffplug.gradle.spotless' version '6.11.0' apply false
    id 'org.jetbrains.kotlin.android' version '1.6.21' apply false
}
```

File: code/settings.gradle [hunk 12]
```diff
@@ -1,3 +1,43 @@
...
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        mavenCentral()
+//        maven { url "https://plugins.gradle.org/m2/" }
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+    }
+
+//    apply plugin: 'maven-publish'
```
Comment (kevinlind) [inline code/settings.gradle:38]: remove

**Review by kevinlind — APPROVED** [review]

**Review by cacheung — APPROVED** [review]

**Pull request merged**